packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -234,21 +318,29 @@ const writeRecords = async (cumulusMessage, knex) => {
     return writeRecordsToDynamoDb(cumulusMessage);
   }
 
-  const provider = await getMessageProvider(cumulusMessage, knex);
+  const providerCumulusId = await getMessageProviderCumulusId(cumulusMessage, knex);
 
   try {
-    await writeExecution({
+    const executionCumulusId = await writeExecution({
       cumulusMessage,
       knex,
     });
     // PDR write only attempted if execution saved
     const pdrCumulusId = await writePdr({
       cumulusMessage,
-      collection,
-      provider,
+      collectionCumulusId,
+      providerCumulusId,
+      knex,
+      executionCumulusId,
+    });
+    return await writeGranules({
```

Comment:
Re-evalute logic based on whether there is a PDR in the message. Granule is not dependent on PDR if there isn't one

Reply:
Do you mind elaborating what a PDR/ polling with delivery record mechanism is? Why was this re-written from `writePdr` to `writeGranules`?

Reply:
I think this is two separate questions. I can't fully elaborate on what a PDR is or how it works - but it's some kind of file format that is published by a provider that can be used to notify a DAAC that data is ready for ingest. So we have DiscoverPdrs workflows that discover PDRs on the provider, parse them, and ingest the referenced data. Is that correct @Jkovarik @nemreid ?

Also, this function wasn't rewritten from `writePdr` to `writeGranules`. `writePdr` still exists right above this function call. I did rename `saveGranulesToDb` to `writeGranules` to be consistent in the function naming with how the rest of the code was refactored

Reply:
Also, I took another look at this and I think the logic here is fine. The key thing is that `writeGranules` is not attempted if `writePdr` throws an error, which would only happen if **there is a PDR on the message and writing it somehow fails**. If there is no PDR on the message, `writePdr` just returns `undefined` and `writeGranules` will continue as normal without a reference to a PDR.

Reply:
> Also, this function wasn't rewritten from writePdr to writeGranules. writePdr still exists right above this function call.

You're right, sorry I missed that. And yeah, the logic is solid